### ISSUE TITLE
Import in docker doesn't import all files

### ISSUE THREAD

**Issue by zinz-code** [photoprism/photoprism#1453]
I am using photoprism on docker on Unraid.
For some reason the import doesn't import all the pictures in the import directory.
There are around 330k pictures in there without duplicates. Currently there are around 230k imported.
To reach that I had to start the importer several times.

What settings do I need so that the import runs until it is actually finished importing?

Logfile:
> time="2021-07-26T15:31:27-07:00" level=info msg="media: 20180923_133515.jpg was taken at 2018-09-23 13:35:15 +0000 UTC (meta)"
> time="2021-07-26T15:31:27-07:00" level=info msg="media: Screenshot_2018-09-05-18-49-59.png was taken at 2018-09-05 18:49:59 +0000 UTC (name)"
> time="2021-07-26T15:31:27-07:00" level=info msg="import: Screenshot_2018-09-05-18-49-59.png is identical to 2018/09/20180905_184959_E7DA90C6.png (sha1 a0ae7738ef8268d56844a50045674e033b6e1fc0)"
> time="2021-07-26T15:31:27-07:00" level=info msg="import: 20180923_133515.jpg is identical to 2018/09/20180923_133515_C67588CD.jpg (sha1 a352b35d687677175df337a3e321f098293a3512)"
> time="2021-07-26T15:31:27-07:00" level=info msg="media: gif_20180918112947.gif was taken at 2018-11-11 22:24:20 +0000 UTC (file mod time)"
> time="2021-07-26T15:31:27-07:00" level=info msg="import: gif_20180918112947.gif is identical to 2018/11/20181111_222420_04B53F93.gif (sha1 911ecad69add24ec0a7296503cbcf7830f26da9d)"
> time="2021-07-26T15:31:27-07:00" level=info msg="media: Yes They Caan.avi.wmv was taken at 2010-07-08 01:44:50 +0000 UTC (file mod time)"
> time="2021-07-26T15:31:28-07:00" level=info msg="import: “Yes They Caan.avi.wmv” is identical to 2010/07/20100708_014450_AF2F514E.wmv (sha1 4b82ea5ed63e7bc32d4f96af1e12d3b9e47ba4e0)"
> time="2021-07-26T15:31:31-07:00" level=info msg="Import completed in 13239 s"
> time="2021-07-27T04:02:06-07:00" level=info msg="shutting down..."
> time="2021-07-27T04:02:06-07:00" level=info msg="shutting down workers"
> time="2021-07-27T04:02:06-07:00" level=info msg="closed database connection"
> time="2021-07-27T04:02:06-07:00" level=info msg="http: shutting down web server"
> time="2021-07-27T04:02:06-07:00" level=info msg="http: web server shutdown complete"

**Comment by lastzero** [comment]
Seems some of the files were imported before, even though they may be unique in the import folder. Is that possible?

**Comment by zinz-code** [comment]
yeah as stated I started the importer several times. The first time around 100k pictures got imported. I didn't look at the log at that time

**Comment by lastzero** [comment]
Might also be a timing issue as the importer uses multiple workers. Limiting the number of workers to 1 may give you a more deterministic behavior. If there are no duplicates and no already imported files at all, there should be no timing issues in general.

**Comment by zinz-code** [comment]
well so the multithreading is broken?
I am trying with one worker now.

Well something did stop the import process the first time.

**Comment by lastzero** [comment]
No, it's not broken in general... Just said there may be rare timing issues. More likely are memory issues which cause a container restart. Or you files have been added after import has been started. It's not easy to remote debug like that. Note we're on vacation right now.

**Comment by graciousgrey** [comment]
@zinz-code did it work with 1 worker?

**Comment by graciousgrey** [comment]
I will close this due to inactivity.
